q2_demux/_summarize/_visualizer.py

```diff
@@ -231,8 +242,9 @@ def summarize(output_dir: str, data: _PlotQualView, n: int = 10000) -> None:
         json.dump({'n': int(n), 'totalSeqCount': int(sequence_count),
                    'minSeqLen': min_seq_len}, fh)
         fh.write(',')
-        forward_stats.to_json(fh)
-        if paired:
+        if not forward_scores.empty:
```

Comment:
This might break the visualization's [init hook](https://github.com/qiime2/q2-demux/blob/master/q2_demux/_summarize/assets/src/main.js) without a corresponding `else` statement here to ensure that the argument has _some_ value in it.

Reply:
Wouldn't it make more sense to change the init hook to `fwdScores = undefined` than to fill it with a placeholder value?

Reply:
I think "more sense" is a subjective term here --- go with whatever you think works best for you. My concern is this: what happens if badly structured data makes it into this viz? For example, no fwd quality scores, but rev quality scores (I know that example makes no real sense, but, we have seen stranger things). For that example, as you have written it now, there will be too few args for the function, everything will wind up being "offset" in the signature.

Reply:
Very true